### PR TITLE
refactor: update theme values

### LINKED ISSUE #445: Audit Theme Style Variables
Goals:
- Make sure the variable names are logical and obvious
- Make sure the set of variables comprehensive, and balances flexibility with helpful rigidity
- Consider adding spacing variables to replace `spacingScale()`
  - Consider using exponential scale
  - Consider using `rem` values instead of pixels

Define roadmap for any variable name changes, considering any change will impact downstream applications.

### PR DESCRIPTION
**Pull request by shanberg**
Major
- Update theme values in line with those described in #445 
- Fixes and minor refactor to align component styles with proper new values
- Minor updates to font, color, radius values
- NavCardGroup prefers `keenDark`

Minor
- `keenDark` now extends `keen`
- Update Theme/Styles stories to show new values

Closes #445, #465

### CODE REVIEW

**Comment by kaitmore** [comment]
This LGTM after a cursory review of storybook. The only component I wonder about is NavCard - I suppose this might be one of those edge cases we talked about the other day, where the theme needs to be inverted? 

image

**Comment by shanberg** [comment]
@kaitmore Yeah, looks like there's an interesting problem to solve here...

- We can modify the NavCardGroup's default theme, but that gets overridden the moment it’s used in an app with a higher-level `ThemeProvider`, including in Storybook, which makes both unusable _and_ hard to test..
- We can add an inverted set to the `keen` theme but that gets recursive real quick.
- We can add custom un-themed styles for this component alone, but that's kind of a nightmare.
- We can import the dark theme and use it directly so that we benefit from the real values, but then we can't modify it based on context later.
- ...Better ideas?

**Comment by shanberg** [comment]
useContextTheme

How about this?

```
function NavCardGroup({ useContextTheme, children }) {
  return (
    <ThemeProvider theme={props => (useContextTheme ? props : keenDark)}>
      <NavCardGroupWrap>{children}</NavCardGroupWrap>
    </ThemeProvider>
  );
}```

**Comment by shanberg** [comment]
And thus is born a new React pattern.

**Comment by dance-dance-banana-frenzy** [comment]
@shanberg we need a meeting of the minds with this in regards to context menu and drop-down menu in `gm-data` repo since they use `COLOR_BACKGROUND_C` instead of a separate theme. The date-picker field is going to run into this issue as well.

File: README.md [hunk 1]
```diff
@@ -97,67 +97,87 @@ To extend the theme, you must pass a theme object that follows our schema. Here
...
-  COLOR_BRAND_A: "#0aab2a",
-  COLOR_BRAND_B: "#0aab2a",
-  COLOR_CONTENT: "#000",
+  COLOR_BRAND_PRIMARY: "#00b42b",
+  COLOR_BRAND_SECONDARY: "#00b42b",
 
-  COLOR_BACKGROUND_A: "#ffffff",
-  COLOR_BACKGROUND_B: "#F8F8F8",
-  COLOR_BACKGROUND_C: "#000000",
+  COLOR_BACKGROUND_DEFAULT: "hsl(0,0%,100%)",
```
Comment (dance-dance-banana-frenzy) [inline README.md]: These are breaking changes, usually resulting in a major version update in regards to release management and versioning. It's going to impact `gm-ui` everywhere that we use the old keys.

Comment (shanberg) [reply README.md]: Yep, we understand.

The theme value transformations are recorded [here](https://gist.github.com/shanberg/29a62d39d4c08a42dbcfcec7de91be27), and moving to 1.0 will involve a little hands-on work which I plan to be fully available for.

**Pull request merged**